src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -88,7 +88,7 @@ void Ekf::controlGpsFusion()
 							// use GPS velocity data to check and correct yaw angle if a FW vehicle
 							if (_control_status.flags.fixed_wing && _control_status.flags.in_air) {
 								// if flying a fixed wing aircraft, do a complete reset that includes yaw
-								_control_status.flags.mag_aligned_in_flight = realignYawGPS();
+								_mag_yaw_reset_req = true;
```

Comment:
Is this acceptable? The potential difference is that `realignYawGPS()` won't be called until a mag sample is ready.

Reply:
I think it's ok for now, but I see that the problem is in that `realignYawGPS` that does a bit everything. It should just try to realign using the GPS measurements and return false if it doesn't work; If it doesn't work, then set `_mag_yaw_reset_req = true` and let the "mag control logic" do it's job.

It requires a bit more refactoring so we can leave it like that and do it later.